pkg/policyviolation/clusterpv.go

```diff
@@ -0,0 +1,152 @@
+package policyviolation
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/golang/glog"
+	kyverno "github.com/nirmata/kyverno/pkg/api/kyverno/v1"
+	kyvernov1 "github.com/nirmata/kyverno/pkg/client/clientset/versioned/typed/kyverno/v1"
+	kyvernolister "github.com/nirmata/kyverno/pkg/client/listers/kyverno/v1"
+	client "github.com/nirmata/kyverno/pkg/dclient"
+	labels "k8s.io/apimachinery/pkg/labels"
+)
+
+func createPVS(dclient *client.Client, pvs []kyverno.ClusterPolicyViolation, pvLister kyvernolister.ClusterPolicyViolationLister, pvInterface kyvernov1.KyvernoV1Interface) error {
+	for _, pv := range pvs {
+		if err := createPVNew(dclient, pv, pvLister, pvInterface); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (gen *Generator) createCusterPV(info Info) error {
+	var pvs []kyverno.ClusterPolicyViolation
+	if !info.Blocked {
+		pvs = append(pvs, buildPV(info))
+	} else {
+		// blocked
+		// get owners
+		pvs = buildPVWithOwners(gen.dclient, info)
+	}
+	// create policy violation
+	if err := createPVS(gen.dclient, pvs, gen.pvLister, gen.pvInterface); err != nil {
+		return err
+	}
+
+	glog.V(3).Infof("Created cluster policy violation policy=%s, resource=%s/%s/%s",
+		info.PolicyName, info.Resource.GetKind(), info.Resource.GetNamespace(), info.Resource.GetName())
+	return nil
+
+}
+func createPVNew(dclient *client.Client, pv kyverno.ClusterPolicyViolation, pvLister kyvernolister.ClusterPolicyViolationLister, pvInterface kyvernov1.KyvernoV1Interface) error {
+	var err error
+	// PV already exists
+	ePV, err := getExistingPVIfAny(pvLister, pv)
+	if err != nil {
+		glog.Error(err)
+		return fmt.Errorf("failed to get existing pv on resource '%s': %v", pv.Spec.ResourceSpec.ToKey(), err)
```

Comment:
As previously we are printing out kind/namespace/name, would be nice to print out the namespace here as well.